### PR TITLE
ImPlement lowering for normal Op

### PR DESCRIPTION
**Pull request by JackCaoG**

### CODE REVIEW

File: torch_xla/csrc/xla_lower_util.cpp [hunk 1]
```diff
@@ -642,6 +643,11 @@ std::vector<xla::XlaOp> BuildNonZero(xla::XlaOp input) {
       xla::Ne(input, xla::Zero(input.builder(), input_shape.element_type())));
 }
 
+xla::XlaOp BuildNormal(xla::XlaOp mean, xla::XlaOp std, xla::XlaOp rng_seed) {
```
Comment (dlibenzi) [inline torch_xla/csrc/xla_lower_util.cpp]: No need for this 2 liner. Expand at call site.

File: torch_xla/csrc/xla_lower_util.h [hunk 2]
```diff
@@ -58,6 +58,8 @@ xla::XlaOp CreatePut(xla::XlaOp input, xla::XlaOp index, xla::XlaOp source,
 
 std::vector<xla::XlaOp> BuildNonZero(xla::XlaOp input);
 
+xla::XlaOp BuildNormal(xla::XlaOp mean, xla::XlaOp std, xla::XlaOp rng_seed);
```
Comment (dlibenzi) [inline torch_xla/csrc/xla_lower_util.h]: No need for this API. Expand at call site.

File: torch_xla/csrc/ops/normal.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    return BuildNormal(operands[0], operands[1], rng_seed);
+  };
+  return InferOutputShape({mean.shape(), std.shape()}, lower_for_shape_fn);
+}
+
+}  // namespace
+
+Normal::Normal(const Value& mean, const Value& std, xla::uint64 seed)
+    : Node(ir::OpKind(at::aten::normal), {mean, std},
+           [&]() { return NodeOutputShape(mean, std, seed); },
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/normal.cpp]: You can pass mean.shape() here instead of inferring.

File: torch_xla/csrc/ops/normal.cpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
+#include "torch_xla/csrc/ops/normal.h"
+
+#include "absl/strings/str_join.h"
+#include "tensorflow/compiler/xla/xla_client/util.h"
+#include "torch_xla/csrc/helpers.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/normal.cpp]: No need.

File: torch_xla/csrc/ops/normal.cpp [hunk 5]
```diff
@@ -0,0 +1,54 @@
+#include "torch_xla/csrc/ops/normal.h"
+
+#include "absl/strings/str_join.h"
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/normal.cpp]: No need.

File: test/torch_test_meta.py [hunk 6]
```diff
@@ -155,6 +155,7 @@
     'test_triangular_solve',  # (TPU) precision (1e-7)
     'test_scalar_check',  # runtime error
     'test_argminmax_large_axis',  # OOM, and the test is grepping "memory" in the exception message
+    'test_trapz' # precision (1e-5)
```
Comment (dlibenzi) [inline test/torch_test_meta.py]: I tested this right now, and the test is green.
Seem strange as the default precision is:
```Python
DEFAULT_FLOATING_PRECISION = 1e-3
```
In `torch_test_meta.py`.
Maybe `test_trapz` uses `normal*()` and this lead to out-of-precision errors.
Can you try to fiddle with `torch_test_precisions` in `torch_test_meta.py`?
Even though the 1e-3 default should do it ... assuming it actually works.

File: torch_xla/csrc/ops/normal.h [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+class Normal : public Node {
+ public:
+  Normal(const Value& mean, const Value& std, xla::uint64 seed);
+
+  std::string ToString() const override;
+
+  NodePtr Clone(OpList operands) const override;
+
+  XlaOpVector Lower(LoweringContext* loctx) const override;
+
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/normal.h]: I think we should expose the seed.
I know I did not in the rrelu op, but we can change that too.

File: torch_xla/csrc/ops/normal.cpp [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  xla::XlaOp std = loctx->GetOutputOp(operand(1));
+  xla::XlaOp rng_seed =
+      XlaHelpers::ScalarValue(seed_, xla::PrimitiveType::U64, mean.builder());
+  return ReturnOp(BuildNormal(mean, std, rng_seed), loctx);
+}
+
+std::string Normal::ToString() const {
+  std::stringstream ss;
+  ss << Node::ToString();
+  return ss.str();
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/normal.cpp]: Emit seed_.

File: test/torch_test_meta.py (hunk 6 above)
Comment (JackCaoG) [reply test/torch_test_meta.py]: I was wrong about the 1e-5 part, it should be 1e-3.

```
(Pdb) p expected
array([-7.32792318e-03,  1.01249252e+01])
(Pdb) p actual
tensor([-7.3280e-03,  1.0125e+01], device='xla:0')
(Pdb) self.precision
0.001
(Pdb) self.assertEqual(expected, actual.)
(Pdb) self.assertTrue(np.allclose(expected, actual.cpu().numpy()))
AssertionError: False is not true
```

trapz does not use normal, however the test use np.allClose which has the defualt rtol 1e-5. I will update the comment.

Comment (JackCaoG) [reply test/torch_test_meta.py]: Will fix in next pr

**Pull request merged**